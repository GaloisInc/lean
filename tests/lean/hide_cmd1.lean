section tst1
open nat
#check succ
hide succ
#check succ -- Error
end tst1

section tst2
open nat
hide succ
def succ := tt
#check succ -- Should not fail, it is not ambiguous since the alias nat.succ has been hidden
end tst2

section tst3
open nat
section nested
hide zero
#check zero -- Error
end nested
#check zero -- Should work, the scope of the previous `hide` is the section nested
end tst3

namespace tst4
open nat
namespace nested
hide zero
#check zero -- Error
end nested
#check zero -- Should work, the scope of the previous `hide` is the namespace nested
end tst4

section tst5
hide nat.succ -- Error, we can only hide aliases for now
end tst5

section tst6
#check @is_true -- is_true is an alias for decidable.is_true
hide is_true
#check @is_true -- Error
end tst6
#check @is_true

def is_true := tt
#check is_true -- Error, is_true is now ambiguous
hide is_true
#check is_true -- should work, is_true : bool

section tst7
open list
#check map
#check foldl
hide map foldl
#check map -- Error
#check foldl -- Error
end tst7
