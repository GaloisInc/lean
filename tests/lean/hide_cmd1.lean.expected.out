succ : ℕ → ℕ
hide_cmd1.lean:5:7: error: unknown identifier 'succ'
succ : bool
hide_cmd1.lean:19:7: error: unknown identifier 'zero'
0 : ℕ
hide_cmd1.lean:28:7: error: unknown identifier 'zero'
0 : ℕ
hide_cmd1.lean:34:0: error: invalid 'hide' command, 'nat.succ' is not an alias
is_true : Π {p : Prop}, p → decidable p
hide_cmd1.lean:40:8: error: unknown identifier 'is_true'
is_true : Π {p : Prop}, p → decidable p
hide_cmd1.lean:45:7: error: ambiguous overload, possible interpretations
  is_true
  decidable.is_true
Additional information:
hide_cmd1.lean:45:7: context: switched to basic overload resolution where arguments are elaborated without any information about the expected type because expected type was not available
is_true : bool
map : (?M_1 → ?M_2) → list ?M_1 → list ?M_2
foldl : (?M_1 → ?M_2 → ?M_1) → ?M_1 → list ?M_2 → ?M_1
hide_cmd1.lean:54:7: error: unknown identifier 'map'
hide_cmd1.lean:55:7: error: unknown identifier 'foldl'
